core/domain/summary_services.py

```diff
@@ -256,3 +256,19 @@ def get_top_rated_exploration_summary_dicts(language_codes):
         reverse=True)[:NUMBER_OF_TOP_RATED_EXPLORATIONS]
 
     return _get_displayable_exp_summary_dicts(sorted_exp_summaries)
+
+
+def get_recently_published_exploration_summary_dicts(language_codes):
+    """Returns a list of recently published explorations
+     with the given language code.
+    """
+    collect_all_exp_summaries = [
+        exp_summary for exp_summary in
```

Comment:
I'd prefer the first possibility (show all regardless language). This more reflects the `Recently Published` category and it better caters to learners who are multi-lingual. To those who aren't, it lets them know we are an international website.